CMakeLists.txt

```diff
@@ -18,6 +18,29 @@ option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
 
+if(CMAKE_CROSSCOMPILING)
+  set(BOARD_NAME "" CACHE STRING "Enter the board name to compile mlpack for.")
+  set(GFORTRAN_LIBRARY "" CACHE STRING "Path for gfortran library.")
+  set(GCC_VERSION ""  CACHE STRING "")
+endif()
+
+if (CMAKE_CROSSCOMPILING AND (BOARD_NAME STREQUAL ""))
+  message(FATAL_ERROR "Please enter the board name")
+elseif(CMAKE_CROSSCOMPILING AND (BOARD_NAME STRGREATER ""))
+  include(board/flags-config.cmake)
+  flags(${BOARD_NAME})
+
+  set(OPENBLAS_LIBRARIES "${CMAKE_BINARY_DIR}/../OpenBLAS-0.3.13/libopenblas.a" CACHE STRING "OpenBLAS PATH")
```

Comment:
Maybe I missed something but does it need to be `openblas`, I thought `blas, lapack` is good as well?

Reply:
Because we have to download them and cross compile them before doing anything, if you would like we can add `blas` and `lapack` but this should mean that we need to add them to the auto-downloader and cross compile them for each architecture, even if at the end we are going to use the best one `OpenBLAS` usually.

Reply:
Auto-downloaders should probably be optional---so if the user wants, they can provide their own statically-cross-compiled dependency.  It might be a little bit more CMake work to get that support, though.

Reply:
I added an optional autodownloader, it should be set by the user if the deps are not available. The default value is set to `OFF`